draft-ietf-quic-recovery.md

```diff
@@ -816,48 +816,92 @@ The minimum congestion window is the smallest value the congestion window can
+The congestion controller has three distinct states, as shown in
+{{fig-cc-fsm}}.
+
+~~~
+                 Start or         +------------+
+            persistent congestion |   Slow     |
+        (O)---------------------->|   Start    |
+                                  +------------+
+                                        |
+                                Loss or |
+                        ECN-CE increase |
+                                        v
+ +------------+     Loss or       +------------+
+ | Congestion |  ECN-CE increase  |  Recovery  |
+ | Avoidance  |------------------>|   Period   |
+ +------------+                   +------------+
+           ^                            |
+           |                            |
+           +----------------------------+
+              Acknowledgment of packet
+                sent during recovery
+~~~
+{: #fig-cc-fsm title="Congestion Control States and Transitions"}
+
+These states and the transitions between them are described in subsequent
+sections.
+
+### Slow Start
+
+A NewReno sender is in slow start any time the congestion window is below the
+slow start threshold. A sender begins in slow start because the slow start
+threshold is initialized to an infinite value.
+
+While a sender is in slow start, the congestion window increases by the number
+of bytes acknowledged when each acknowledgment is processed. This results in
 exponential growth of the congestion window.
 
-A sender MUST exit slow start and enter recovery when a loss is detected or when
-the ECN-CE counter reported by the peer increases.
+The sender exits slow start and enters a recovery period when a packet is lost
+or when the ECN-CE count reported by its peer increases.
 
 A sender re-enters slow start any time the congestion window is less than the
-slow start threshold, which only occurs after persistent congestion is declared.
+slow start threshold, which only occurs after persistent congestion is
+declared.
 
-## Congestion Avoidance
+### Recovery Period {#recovery-period}
 
-A NewReno sender in congestion avoidance uses an Additive Increase
-Multiplicative Decrease (AIMD) approach that MUST increase the congestion window
-by no more than one maximum datagram size per congestion window acknowledged.
+A NewReno sender enters a recovery period when it detects the loss of a
+packet or the ECN-CE count reported by its peer increases. A sender that is
+already in a recovery period stays in it and does not re-enter it.
 
-A sender MUST exit congestion avoidance and enter recovery when a loss is
-detected or when the ECN-CE counter reported by the peer increases.
+On entering a recovery period, a sender MUST set the slow start threshold to
+half the value of the congestion window at the moment that loss is
+detected. The congestion window MUST be set to the reduced value of the slow
+start threshold before exiting the recovery period. Implementations MAY set the
```

Comment:
I prefer the organization the existing text for this paragraph, and starting the new paragraph with "Implementations MAY..."

Reply:
Just a consequence of many iterations of merging and rebasing.  Fixed.